server/src/main/scala/za/co/absa/atum/server/api/database/DoobieImplicits.scala

```diff
@@ -42,15 +43,22 @@ object DoobieImplicits {
 
   object Json {
 
-    implicit val jsonArrayPutUsingString: Put[List[String]] = {
+    implicit val jsonArrayPut: Put[List[Json]] = {
       Put.Advanced
         .other[PGobject](
           NonEmptyList.of("json[]")
         )
         .tcontramap { a =>
           val o = new PGobject
           o.setType("json[]")
-          o.setValue(a.mkString("{", ",", "}"))
+          val arrayElements = a.map { x =>
```

Comment:
You could define private method in DoobieImplicits object as below in order to avoid repetition.

```scala
private def circeJsonListToPGJsonArrayString(jsonList: List[Json]): String = {
    val arrayElements = jsonList.map { x =>
      // Convert to compact JSON string and escape inner quotes
      val escapedJsonString = x.noSpaces.replace("\"", "\\\"")
      // Wrap in double quotes for the array element
      s"\"$escapedJsonString\""
    }

    // Join all elements into a single string wrapped in curly braces
    arrayElements.mkString("{", ",", "}")
  }
```

Usage:

```scala
implicit val jsonArrayPut: Put[List[Json]] = {
      Put.Advanced
        .other[PGobject](
          NonEmptyList.of("json[]")
        )
        .tcontramap { a =>
          val o = new PGobject
          o.setType("json[]")
          o.setValue(circeJsonListToPGJsonArrayString(a))
          o
        }
    }
```